src/trace/descriptor.ts

```diff
@@ -36,7 +36,17 @@ export function augmentLogs(logs: StructLog[], constants: object): { logs: Log[]
     const nextStack = nextLog ? nextLog.stack : undefined;
 
     let { extended, info: nInfo } = describeOperation(log, nextStack || [], info);
-    const nLog: Log = new Log(log, extended, info.lastLog);
+    let nLog: Log = new Log(log, extended, info.lastLog);
+
+    if (nextLog != undefined) {
+      nLog.gasCost = nextLog.gasCost;
+    }
+
+    if (["STATICCALL", "DELEGATECALL", "CALL"].includes(log.op)) {
```

Comment:
This is great-- it'd be cool one day to try and make it even smarter, possibly by iterating through again and matching returns to calls, etc, and seeing if there's more analysis we can, but this is a great fix.

Reply:
good point, just added this